### DISCUSSION TITLE
MxNet unable to detect CPU on AWS/EMR/g4 instance

### DISCUSSION THREAD

**Discussion by patrick-nicolas** [deepjavalibrary/djl#2038] [General]
The application works just fine on CPU. I followed the recommendations from issue 732
[Issue-732](https://github.com/deepjavalibrary/djl/discussions/732)
Which step am I missing?

**GPU configuration:**
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 450.172.01   Driver Version: 450.172.01   CUDA Version: 11.0     |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|                               |                      |               MIG M. |
|===============================+======================+======================|
|   0  Tesla T4            Off  | 00000000:00:1E.0 Off |                    0 |
| N/A   33C    P0    14W /  70W |      0MiB / 15109MiB |      0%      Default |
|                               |                      |                  N/A |
+-------------------------------+----------------------+----------------------+
+-----------------------------------------------------------------------------+
| Processes:                                                                  |
|  GPU   GI   CI        PID   Type   Process name                  GPU Memory |
|        ID   ID                                                   Usage      |
|=============================================================================|
|  No running processes found                                                 |

**MxNet binary libraries**
_/home/hadoop/.djl.ai/mxnet/1.9.0-mkl-linux-x86_64/
├── libgfortran.so.3
├── libgomp.so.1
├── libmxnet.so
├── libopenblas.so.0
└── libquadmath.so.0_

**Relevant environment Variables**
PATH: /home/hadoop/spark-3.3.0-bin-hadoop3-scala2.13/bin/:/home/hadoop/jdk-11/bin:/usr/local/cuda/bin:/usr/local/cuda/lib64:usr/local/cuda/targets/x86_64-linux/lib:/usr/lib64/qt-3.3/bin:/usr/local/bin:/usr/bin:/usr/local/sbin:/usr/sbin:/opt/aws/puppet/bin/
LD_LIBRARY_PATH: /usr/local/cuda/lib64:/usr/local/cuda/targets/x86_64-linux/lib
JAVA_HOME: /home/hadoop/jdk-11
SPARK_ENV_LOADED: 1
SPARK_SCALA_VERSION: 2.13
SPARK_HOME: /home/hadoop/spark-3.3.0-bin-hadoop3-scala2.13
SHELL: /bin/bash
USER: hadoop
SPARK_CONF_DIR: /home/hadoop/spark-3.3.0-bin-hadoop3-scala2.13/conf
HOME: /home/hadoop

-------------- Directories --------------
temp directory: /tmp
DJL cache directory: /home/hadoop/.djl.ai
Engine cache directory: /home/hadoop/.djl.ai

------------------ CUDA -----------------
GPU Count: 1
CUDA: 110
ARCH: 75
GPU(0) memory used: 106954752 bytes

----------------- Engines ---------------
MXNet version: 1.9.0
Default Engine: MXNet
Default Device: cpu()
MXNet:  #0

**Comment by frankfliu** [comment]
@patrick-nicolas 
MXNet 1.9.1 only support CUDA 11.2/10.2, see: https://docs.djl.ai/engines/mxnet/mxnet-engine/index.html#linux-gpu
